string_match_without_scan_or_regex_spec.rb

```diff
@@ -0,0 +1,144 @@
+require_relative 'string_match_without_scan_or_regex'
+
+describe String do
+
+  let(:randomly_generated_strings){ (0..2).map{(0..rand(1..100)).map{(rand(127)).chr}.join}}
+  let(:string){ randomly_generated_strings[0] + "like" + randomly_generated_strings[1] + "like" + randomly_generated_strings[2]}
+  let(:substring){"like"}
+  let(:substring_two){"nope"}
+
+
+  #Testing to ensure above variables used in the spec are objects of the class String
+  describe 'string variables' do
+    describe 'string' do
+      it 'should be an object of the class String' do
+        expect(string).to be_kind_of(String)
+      end
+    end
+
+    describe 'substring' do
+      it 'should be an object of the class String' do
+        expect(substring).to be_kind_of(String)
+      end
+    end
+
+    describe 'substring_two' do
+      it 'should be an object of the class String' do
+        expect(substring_two).to be_kind_of(String)
+      end
+    end
+  end
+
+
+  #Testing #match? functionality for objects of the class String
+  describe '#match?' do
+
+    it 'should be a defined method on the class String' do
+      expect(String.method_defined?(:match?)).to be(true)
+      #a test against false positives for defined methods
+      expect(String.method_defined?(:matchez?)).to be(false)
+    end
+
+    it 'will raise an error if called on an instance that is not a String object' do
+      expect{[].match?(string)}.to raise_error(NoMethodError)
+      expect{1.match?(string)}.to raise_error(NoMethodError)
+      expect{{}.match?(string)}.to raise_error(NoMethodError)
+    end
+
+    it 'expects a single argument' do
+      expect(string.method(:match?).arity).to eq(1)
```

Comment:
These arity checks are just going to cause you pain.  I love the thoroughness, but it restricts your ability to change code, and doesn't really test any behavior.  Imagine if you need to add an argument, how much extra work it is to change these tests.